Spatial.gd

```diff
@@ -413,13 +420,81 @@ func gen_map_markers():
 		var position_vector = Vector3(position.get_x(), position.get_y(), position.get_z())
 		var texture_path = icon.get_texture_path()
 		if texture_path == null:
-			print("Warning: No texture found for icon")
+			#Some icons have their texture in a parent of the category.
 			continue
 		var full_texture_path = self.marker_file_dir + texture_path.get_path()
-		gen_new_icon(position_vector, full_texture_path)
+		gen_new_icon(position_vector, full_texture_path, icon.get_category().get_name())
+	
+		
+onready var local_data = get_node("Control/LocalData") 
+
+func build_category_tree():
+	var root = marker_packs.create_item()
+	root.set_text(0, "Markers available on current map")
+	root.set_selectable(0, false)
+	root.set_text(1, "Visible")
+	
+	for category in self.markerdata.get_category():
+		add_category(root, category, category.get_name(), false)
+
+
+func add_category(item: TreeItem, category, tree_path: String, collapsed: bool):
+	if category.get_name() != "":
```

Comment:
Should there be an error if the category name is empty? Could there be nodes we are not showing because of an empty category name?

If this IS correct then you should use

```gdscript
if category.get_name() == "":
    return
```

Reply:
Added a warning as well